sw-services/src/main/java/org/egov/swservice/service/SewerageFieldValidator.java

```diff
@@ -13,32 +13,80 @@
 public class SewerageFieldValidator implements SewerageActionValidator {
 
 	@Override
-	public ValidatorResult validate(SewerageConnectionRequest sewerageConnectionRequest, boolean isUpdate) {
+	public ValidatorResult validate(SewerageConnectionRequest sewerageConnectionRequest, int reqType) {
 		Map<String, String> errorMap = new HashMap<>();
-		if (isUpdate) {
-			if (SWConstants.ACTIVATE_CONNECTION_CONST.equalsIgnoreCase(
-					sewerageConnectionRequest.getSewerageConnection().getProcessInstance().getAction())) {
-				if (StringUtils.isEmpty(sewerageConnectionRequest.getSewerageConnection().getConnectionType())) {
-					errorMap.put("INVALID_SEWERAGE_CONNECTION_TYPE", "Connection type should not be empty");
-				}
-				if (StringUtils
-						.isEmpty(sewerageConnectionRequest.getSewerageConnection().getConnectionExecutionDate())) {
-					errorMap.put("INVALID_CONNECTION_EXECUTION_DATE", "Connection execution date should not be empty");
-				}
+		switch (reqType){
+			case SWConstants.UPDATE_APPLICATION:
+				validateUpdateRequest(sewerageConnectionRequest, errorMap);
+			    break;
+			case SWConstants.MODIFY_CONNECTION:
+				validateModifyRequest(sewerageConnectionRequest, errorMap);
+			default:
+				break;
+		}
+		if (!errorMap.isEmpty())
+			return new ValidatorResult(false, errorMap);
+		return new ValidatorResult(true, errorMap);
+	}
+
+	public void validateUpdateRequest(SewerageConnectionRequest sewerageConnectionRequest, Map<String, String> errorMap) {
+		if (SWConstants.ACTIVATE_CONNECTION_CONST.equalsIgnoreCase(
+				sewerageConnectionRequest.getSewerageConnection().getProcessInstance().getAction())) {
```

Comment:
is the application searched in DB for validation done somewhere else?

Reply:
No, It will come from the request.